pages/signin/index.tsx

```diff
@@ -1,13 +1,86 @@
 import { Button } from '../../components/buttons';
+import { Card, CardContent, Typography, Container } from '@material-ui/core';
 import { Layout } from '../../containers/layout';
+import InputField from '../../components/input-field/InputField';
+import { useState } from 'react';
+import { PageProps, UserInput } from '../../lib/types';
+import { withTranslation } from '../../i18n';
+import styles from './Signin.module.css';
+import { useAuth } from '../../hooks/useAuth';
+import { FirebaseUser } from '../../lib/types/auth';
+
+const SignInPage = ({ t }: PageProps): React.ReactElement => {
+  const [formValues, setFormValues] = useState<Partial<UserInput>>({
```

Comment:
So you mean e.g. for auth.ts export default an object that contains the Signup, Signin, Logout, FirebaseUser etc. types?
So that you import authTypes and then use authTypes.Signup ?

This could be an option. But not sure if this makes it better. In the end you have to know which type you need.

Reply:
no, the default objects. E.g.
defaultUserInput = {
lastName: '',
firstName: '',
...
}
This way, we have types and default instances in one place (in the types file). This is convenient, since we need default instances because graphql mutations expect them.

Reply:
not particularly pertinent to this pr, however. But we should discuss, decide, and then create a ticket (if we want it).